### ISSUE TITLE
The left vertical line highlighting the scope of the current function does not work properly 

### ISSUE THREAD

**Issue by jansviland** [izhangzhihao/intellij-rainbow-brackets#2581]
## Your programming languages

C#

## Expected Behavior

The entire function has a left line showing the scope. 

## Current Behavior

Only part of the scope is highlighted, sometimes it takes some time before it is properly shown. 

## Code snippet for reproduce (for bugs)

<img width="467" alt="Screenshot 2023-01-09 at 11 07 31" src="https://user-images.githubusercontent.com/114911422/211285000-9adb2ca3-66cf-4140-a4e2-f7e8eb066d42.png">

<img width="215" alt="Screenshot 2023-01-09 at 11 13 52" src="https://user-images.githubusercontent.com/114911422/211285212-6e5e31a6-a868-4fb8-912a-d062711d532e.png">

## Your Environment

```
JetBrains Rider 2022.3.1
Build #RD-223.8214.53, built on December 20, 2022
Licensed to XXX
Subscription is active until May 24, 2023.
Runtime version: 17.0.5+1-b653.23 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 13.1
.NET 7.0.0 (Server GC)
GC: G1 Young Generation, G1 Old Generation
Memory: 1504M
Cores: 10
Metal Rendering is ON
Registry:
    ide.lookup.shrink=true
    ide.experimental.ui=true
    indexing.enable.entity.provider.based.indexing=false
    ide.workspace.model.rbs.as.tree=false
    platform.projectModel.workspace.model.file.index=false
    database.show.search.tab=false
    vcs.empty.toolwindow.show=false
    ide.new.project.model.index.case.sensitivity=true
    rdclient.asyncActions=false

Non-Bundled Plugins:
    com.markskelton.one-dark-theme (5.7.2)
    fr.socolin.application-insights-debug-log-viewer (1.7.3)
    com.intellij.resharper.HeapAllocationsViewer (2022.3.1)
    me.seclerp.rider.plugins.efcore (223.2.0)
    com.samdark.intellij-visual-studio-code-dark-plus (2.6)
    com.intellij.resharper.StructuredLogging (2022.3.1.280)
    IdeaVIM (2.0.0)
    com.github.copilot (1.1.38.2229)
    com.intellij.resharper.azure (3.50.0.1539-2022.3)
    izhangzhihao.rainbow.brackets (2023.1.0)



IntelliJ IDEA 2021.2 (Ultimate Edition)
Build #IU-212.4746.92, built on July 27, 2021
Licensed to IntelliJ Rainbow Brackets / Zhang Zhihao
Subscription is active until February 3, 2022.
For non-commercial open source development only.
Runtime version: 11.0.11+9-b1504.13 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 11.4
GC: G1 Young Generation, G1 Old Generation
Memory: 4096M
Cores: 16
Registry: ide.tooltip.initialDelay=900, ide.balloon.shadow.size=0, scala.erase.compiler.process.jdk.once=false
Non-Bundled Plugins: com.intellij.properties.bundle.editor (212.4746.57), com.markskelton.one-dark-theme (5.3.0), lermitage.intellij.extra.icons (1.59.0.203), org.nik.presentation-assistant (1.0.9), kotest-plugin-intellij (1.1.36-IC-2021.1), Pythonid (212.4746.96), org.intellij.scala (2021.2.15), izhangzhihao.rainbow.brackets (6.19), com.intellij.bigdatatools (212.4037.55)
Kotlin: 212-1.5.10-release-IJ4746.92
```

**Comment by izhangzhihao** [comment]
Same with #2540

**Comment by izhangzhihao** [comment]
merged into #2540
